CharlieBackend.Api/Controllers/SecretariesController.cs

```diff
@@ -51,13 +51,28 @@ public async Task<ActionResult> PutSecretary(long secretaryId, UpdateSecretaryDt
             return updatedSecretary.ToActionResult();
         }
 
+
+
         [Authorize(Roles = "Admin")]
         [HttpDelete("{secretaryId}")]
         public async Task<ActionResult> DisableSecretary(long secretaryId)
         {
-            var isDisabled = await _secretaryService.DisableSecretaryAsync(secretaryId);
 
-            return isDisabled.ToActionResult();
+            var accountId = await _secretaryService.GetAccountId(secretaryId);
+
+            if (accountId == null)
+            {
+                return BadRequest("Unknown secretary id.");
+            }
+
+            var isDisabled = await _accountService.DisableAccountAsync((long)accountId);
+
+            if (isDisabled)
+            {
+                return Ok(await _secretaryService.GetSecretaryByIdAsync(secretaryId));
+            }
```

Comment:
Maybe better to have it in service?

Reply:
Better move this logic to service and use Result T